Pull Request Angola 2011 JW 20210531

jianingwwww opened the pull request:
- change the file name, from DHS to MIS
- change the path to ensure it runs successfully (when using "birth" there will be an error "v001 not found")

jianingwwww committed:
file name change

- change the file name, from DHS to MIS
-change the path name to ensure it runs smoothly (when using "birth" there will be an error "v001 not found")

robin-wang commented:
Changed Pull Request Name to reflect current work.

wengxyu1030 commented on MIS_Recode.do:
```diff
@@ -157,10 +157,10 @@ save `hm',replace
     rename (hv001 hv002 hvidx) (v001 v002 v003)
 
-    merge 1:m v001 v002 v003 using `birth'
+    merge 1:m v001 v002 v003 using "${SOURCE}/MIS-`name'/MIS-`name'birth.dta"
```
Could you please clarify why changing the merge data source? Here the command is trying to merge the temp data generated in command line 106. You are changing this to the raw data instead of using the data generated in the previous section of the code. @imarabbit

jianingwwww replied:
I changed the merge data source because when I tried merging the temp data, an error "variable v001 not found" is reported. I have checked the do file and the data but I was not able to locate the problem and solve it (variable v001 is supposed to exist after variable hv001 in the birth temp data being renamed as v001). So I had to change the source to ensure other indicators besides those relating to household level can be generated successfully,  and then reported this problem in the pull request description. This problem also appears when I deal with Angola 2006-07 dataset. Could you please show me how to fix it? Thanks a lot in advance! @wengxyu1030

jianingwwww replied:
> I changed the merge data source because when I tried merging the temp data, an error "variable v001 not found" is reported. I have checked the do file and the data but I was not able to locate the problem and solve it (variable v001 is supposed to exist after variable hv001 in the birth temp data being renamed as v001). So I had to change the source to ensure other indicators besides those relating to household level can be generated successfully, and then reported this problem in the pull request description. This problem also appears when I deal with Angola 2006-07 dataset. Could you please show me how to fix it? Thanks a lot in advance! @wengxyu1030

Sorry to interrupt. I've found the problem, it's because the birth temp data was saved with changes to variable names (from v001 to hv001). I will rework the codes and pull the requests again.

wengxyu1030 replied:
Thanks @imarabbit please @ me here once the modification is completed.

wengxyu1030 replied:
Dear @Xian152. 

Could you kindly review the template changing here in terms of the merging error? I am wondering the modification made by @imarabbit is survey specific or is applying to all the MIS surveys. When you were building the template, did you encounter this same issue for the MIS survey you worked on?

jianingwwww committed:
delete rename lines

- when the birth temp data was saved (in line 106), variables v001 v002 b16 have already been renamed as hv001 hv002 hvidx, so there is no need to rename these three variables in the hm.dta ; otherwise it will report error variable v001 not found

jianingwwww commented:
I have restored the original data source "birth" and deleted the rename lines to avoid the error. Detailed explanation can be seen in the commit description. @wengxyu1030

wengxyu1030 commented:
> I have restored the original data source "birth" and deleted the rename lines to avoid the error. Detailed explanation can be seen in the commit description. @wengxyu1030

Thanks @imarabbit , it's a nice catch. I am inviting another WB team member to look into this to see if this apply to all the surveys in MIS or it's a surey-specific issue.

Xian152 commented:
@wengxyu1030  @imarabbit 
Hi both, first this is a country specific case. XXXXbirth.dta is not avaliable for all survey, which is why we generate a `birthind` data from ind.dta in place of birth.dta. 
As for Anglola2011, though it has the original birth.dta, I found reshaping ind.dta can yield as much needed information as provided by birth.dta, thus I keep this structure for Anglola2011. But it would be nice if you could have a double check. i.e. if generated datafile`birthind` has much needed information as birth.dta. 
As for this situation captured by @imarabbit, yes you are right. The renaming process was missed up. However, since `using "XXXbirth.dta"` doesn't apply to all survey, I would suggest replacing the code with 
```
use "${SOURCE}/DHS-`name'/DHS-`name'hm.dta", clear
    merge 1:m hv001 hv002 hvidx using `birth'
    drop _merge

    do "${DO}/15_household"

keep hv001 hv002 hv003 hh_* ind_*
save `hh' 
```

wengxyu1030 commented:
> @wengxyu1030 @imarabbit
> Hi both, first this is a country specific case. XXXXbirth.dta is not avaliable for all survey, which is why we generate a `birthind` data from ind.dta in place of birth.dta.
> As for Anglola2011, though it has the original birth.dta, I found reshaping ind.dta can yield as much needed information as provided by birth.dta, thus I keep this structure for Anglola2011. But it would be nice if you could have a double check. i.e. if generated datafile`birthind` has much needed information as birth.dta.
> As for this situation captured by @imarabbit, yes you are right. The renaming process was missed up. However, since `using "XXXbirth.dta"` doesn't apply to all survey, I would suggest replacing the code with
> 
> ```
> use "${SOURCE}/DHS-`name'/DHS-`name'hm.dta", clear
>     merge 1:m hv001 hv002 hvidx using `birth'
>     drop _merge
> 
>     do "${DO}/15_household"
> 
> keep hv001 hv002 hv003 hh_* ind_*
> save `hh' 
> ```

Hi @Xian152 

Thanks for your comment. yes, using this code would work for this survey. I am wondering will this code apply to all/most the MIS surveys? Should we make the code conditional on the survey instead of changing it completely if the previous template worked fine on certain surveys (as you were working on certain MIS surveys building the template)?

dkuhds commented:
> > @wengxyu1030 @imarabbit
> > Hi both, first this is a country specific case. XXXXbirth.dta is not avaliable for all survey, which is why we generate a `birthind` data from ind.dta in place of birth.dta.
> > As for Anglola2011, though it has the original birth.dta, I found reshaping ind.dta can yield as much needed information as provided by birth.dta, thus I keep this structure for Anglola2011. But it would be nice if you could have a double check. i.e. if generated datafile`birthind` has much needed information as birth.dta.
> > As for this situation captured by @imarabbit, yes you are right. The renaming process was missed up. However, since `using "XXXbirth.dta"` doesn't apply to all survey, I would suggest replacing the code with
> > ```
> > use "${SOURCE}/DHS-`name'/DHS-`name'hm.dta", clear
> >     merge 1:m hv001 hv002 hvidx using `birth'
> >     drop _merge
> > 
> >     do "${DO}/15_household"
> > 
> > keep hv001 hv002 hv003 hh_* ind_*
> > save `hh' 
> > ```
> 
> Hi @Xian152
> 
> Thanks for your comment. yes, using this code would work for this survey. I am wondering will this code apply to all/most the MIS surveys? Should we make the code conditional on the survey instead of changing it completely if the previous template worked fine on certain surveys (as you were working on certain MIS surveys building the template)?

oooh, allow me to clarify:
The code I give here should apply to other survey. 
The code used by @imarabbit applies to Anglola2011, those surveys that have birth.dta, but doesn't apply to most of the surveys in MIS:
`merge 1:m v001 v002 v003 using "${SOURCE}/MIS-`name'/MIS-`name'birth.dta" `
This is why I am suggesting using the code I gave here.

Xian152 commented:
> > @wengxyu1030 @imarabbit
> > Hi both, first this is a country specific case. XXXXbirth.dta is not avaliable for all survey, which is why we generate a `birthind` data from ind.dta in place of birth.dta.
> > As for Anglola2011, though it has the original birth.dta, I found reshaping ind.dta can yield as much needed information as provided by birth.dta, thus I keep this structure for Anglola2011. But it would be nice if you could have a double check. i.e. if generated datafile`birthind` has much needed information as birth.dta.
> > As for this situation captured by @imarabbit, yes you are right. The renaming process was missed up. However, since `using "XXXbirth.dta"` doesn't apply to all survey, I would suggest replacing the code with
> > ```
> > use "${SOURCE}/DHS-`name'/DHS-`name'hm.dta", clear
> >     merge 1:m hv001 hv002 hvidx using `birth'
> >     drop _merge
> > 
> >     do "${DO}/15_household"
> > 
> > keep hv001 hv002 hv003 hh_* ind_*
> > save `hh' 
> > ```
> 
> Hi @Xian152
> 
> Thanks for your comment. yes, using this code would work for this survey. I am wondering will this code apply to all/most the MIS surveys? Should we make the code conditional on the survey instead of changing it completely if the previous template worked fine on certain surveys (as you were working on certain MIS surveys building the template)?

oooh, allow me to clarify:
The code I give should apply to most of the survey. 
The code used by @imarabbit here applies to Anglola2011, those surveys that have birth.dta, but doesn't apply to most other surveys in MIS:
`merge 1:m v001 v002 v003 using "${SOURCE}/MIS-`name'/MIS-`name'birth.dta" `
This is why I am suggesting using the code I gave in the last reply.
But there is a concern left. I am confident with Anglola2011 that``birthind'` have as much needed information as birth.dta does, so applying the code I gave last time is acceptable. But I am uncertain about other survey that also have birth.dta. If there exist survey where we can generate more variables that we need from birth.dta than ``birthind'`, we will need to make the code conditional on that survey.  @imarabbit  you might need to double check when faced with a survey with birth.dta.

jianingwwww commented:
> > > @wengxyu1030 @imarabbit
> > > Hi both, first this is a country specific case. XXXXbirth.dta is not avaliable for all survey, which is why we generate a `birthind` data from ind.dta in place of birth.dta.
> > > As for Anglola2011, though it has the original birth.dta, I found reshaping ind.dta can yield as much needed information as provided by birth.dta, thus I keep this structure for Anglola2011. But it would be nice if you could have a double check. i.e. if generated datafile`birthind` has much needed information as birth.dta.
> > > As for this situation captured by @imarabbit, yes you are right. The renaming process was missed up. However, since `using "XXXbirth.dta"` doesn't apply to all survey, I would suggest replacing the code with
> > > ```
> > > use "${SOURCE}/DHS-`name'/DHS-`name'hm.dta", clear
> > >     merge 1:m hv001 hv002 hvidx using `birth'
> > >     drop _merge
> > > 
> > >     do "${DO}/15_household"
> > > 
> > > keep hv001 hv002 hv003 hh_* ind_*
> > > save `hh' 
> > > ```
> > 
> > 
> > Hi @Xian152
> > Thanks for your comment. yes, using this code would work for this survey. I am wondering will this code apply to all/most the MIS surveys? Should we make the code conditional on the survey instead of changing it completely if the previous template worked fine on certain surveys (as you were working on certain MIS surveys building the template)?
> 
> oooh, allow me to clarify:
> The code I give should apply to most of the survey.
> The code used by @imarabbit here applies to Anglola2011, those surveys that have birth.dta, but doesn't apply to most other surveys in MIS:
> `merge 1:m v001 v002 v003 using "${SOURCE}/MIS-`name'/MIS-`name'birth.dta" `
> This is why I am suggesting using the code I gave in the last reply.
> But there is a concern left. I am confident with Anglola2011 that`` birthind'` have as much needed information as birth.dta does, so applying the code I gave last time is acceptable. But I am uncertain about other survey that also have birth.dta. If there exist survey where we can generate more variables that we need from birth.dta than ``birthind'`, we will need to make the code conditional on that survey. @imarabbit you might need to double check when faced with a survey with birth.dta.

I have restored the source path as "birth" temp data after I found the renaming error, and replaced the codes with
> > > ```
> > > use "${SOURCE}/MIS-`name'/MIS-`name'hm.dta", clear
> > >     merge 1:m hv001 hv002 hvidx using `birth'
> > >     drop _merge
> > > 
> > >     do "${DO}/15_household"
> > > 
> > > keep hv001 hv002 hv003 hh_* ind_*
> > > save `hh' 
> > > ```

I think this might be the same as your codes given above? Also, I will double check if a survey has birth.dta, thank you for the reminder!

Xian152 commented:
> > > > @wengxyu1030 @imarabbit
> > > > Hi both, first this is a country specific case. XXXXbirth.dta is not avaliable for all survey, which is why we generate a `birthind` data from ind.dta in place of birth.dta.
> > > > As for Anglola2011, though it has the original birth.dta, I found reshaping ind.dta can yield as much needed information as provided by birth.dta, thus I keep this structure for Anglola2011. But it would be nice if you could have a double check. i.e. if generated datafile`birthind` has much needed information as birth.dta.
> > > > As for this situation captured by @imarabbit, yes you are right. The renaming process was missed up. However, since `using "XXXbirth.dta"` doesn't apply to all survey, I would suggest replacing the code with
> > > > ```
> > > > use "${SOURCE}/DHS-`name'/DHS-`name'hm.dta", clear
> > > >     merge 1:m hv001 hv002 hvidx using `birth'
> > > >     drop _merge
> > > > 
> > > >     do "${DO}/15_household"
> > > > 
> > > > keep hv001 hv002 hv003 hh_* ind_*
> > > > save `hh' 
> > > > ```
> > > 
> > > 
> > > Hi @Xian152
> > > Thanks for your comment. yes, using this code would work for this survey. I am wondering will this code apply to all/most the MIS surveys? Should we make the code conditional on the survey instead of changing it completely if the previous template worked fine on certain surveys (as you were working on certain MIS surveys building the template)?
> > 
> > 
> > oooh, allow me to clarify:
> > The code I give should apply to most of the survey.
> > The code used by @imarabbit here applies to Anglola2011, those surveys that have birth.dta, but doesn't apply to most other surveys in MIS:
> > `merge 1:m v001 v002 v003 using "${SOURCE}/MIS-`name'/MIS-`name'birth.dta" `
> > This is why I am suggesting using the code I gave in the last reply.
> > But there is a concern left. I am confident with Anglola2011 that`` birthind'` have as much needed information as birth.dta does, so applying the code I gave last time is acceptable. But I am uncertain about other survey that also have birth.dta. If there exist survey where we can generate more variables that we need from birth.dta than ``birthind'`, we will need to make the code conditional on that survey. @imarabbit you might need to double check when faced with a survey with birth.dta.
> 
> I have restored the source path as "birth" temp data after I found the renaming error, and replaced the codes with
> 
> > > > ```
> > > > use "${SOURCE}/MIS-`name'/MIS-`name'hm.dta", clear
> > > >     merge 1:m hv001 hv002 hvidx using `birth'
> > > >     drop _merge
> > > > 
> > > >     do "${DO}/15_household"
> > > > 
> > > > keep hv001 hv002 hv003 hh_* ind_*
> > > > save `hh' 
> > > > ```
> 
> I think this might be the same as your codes given above? Also, I will double check if a survey has birth.dta, thank you for the reminder!

Yes, it is. Thank you @imarabbit.

wengxyu1030 commented:
Thanks both! 

@Xian152 your explanation makes a lot of sense. Now the template should work for most of the MIS surveys.

And as @imarabbit is currently responsible for all the MIS surveys, I believe once several other MIS surveys coded, she's gonna have more idea of how to treat the cases where birth data is available.

I'll try to run the code again on my end and merge it to the master file if no further issue comes out.
